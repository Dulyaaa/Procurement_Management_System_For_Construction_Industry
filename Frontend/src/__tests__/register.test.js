/* eslint-disable prettier/prettier */
import React from 'react';
import renderer from 'react-test-renderer';
import Register from '../screens/RegisterScreen';

test('Register component renders correctly', () => {
    const tree = renderer.create(<Register />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Should change state if userId is entered', () => {
    const instanceOf = renderer.create(<Register />).getInstance();
    instanceOf.onChangeUserId('Access');
    expect(instanceOf.state.userId).toEqual('Access');
});

it('Should change state if name is entered', () => {
    const instanceOf = renderer.create(<Register />).getInstance();
    instanceOf.onChangeName('Access');
    expect(instanceOf.state.name).toEqual('Access');
});

it('Should change state if email is entered', () => {
    const instanceOf = renderer.create(<Register />).getInstance();
    instanceOf.onChangeEmail('Access');
    expect(instanceOf.state.email).toEqual('Access');
});

it('Should change state if password is entered', () => {
    const instanceOf = renderer.create(<Register />).getInstance();
    instanceOf.onChangePassword('Access');
    expect(instanceOf.state.password).toEqual('Access');
}); 

it('Should change state if role is entered', () => {
    const instanceOf = renderer.create(<Register />).getInstance();
    instanceOf.onChangeRole('Access');
    expect(instanceOf.state.role).toEqual('Access');
});
