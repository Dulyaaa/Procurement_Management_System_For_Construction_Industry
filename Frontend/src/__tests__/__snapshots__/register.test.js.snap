// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#f9f9f9",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView>
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../images/sign.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 100,
            "marginBottom": 35,
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": 350,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E8E8E8",
              "borderRadius": 15,
              "height": 45,
              "marginBottom": 20,
              "width": "70%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Employee ID"
            placeholderTextColor="#003f5c"
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "height": 50,
                "marginLeft": 15,
                "padding": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E8E8E8",
              "borderRadius": 15,
              "height": 45,
              "marginBottom": 20,
              "width": "70%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="User Role"
            placeholderTextColor="#003f5c"
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "height": 50,
                "marginLeft": 15,
                "padding": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E8E8E8",
              "borderRadius": 15,
              "height": 45,
              "marginBottom": 20,
              "width": "70%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Name"
            placeholderTextColor="#003f5c"
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "height": 50,
                "marginLeft": 15,
                "padding": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E8E8E8",
              "borderRadius": 15,
              "height": 45,
              "marginBottom": 20,
              "width": "70%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="User Email"
            placeholderTextColor="#003f5c"
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "height": 50,
                "marginLeft": 15,
                "padding": 10,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E8E8E8",
              "borderRadius": 15,
              "height": 45,
              "marginBottom": 20,
              "width": "70%",
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#003f5c"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "height": 50,
                "marginLeft": 15,
                "padding": 10,
              }
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#FFC400",
            "borderRadius": 25,
            "elevation": 2,
            "height": 45,
            "justifyContent": "center",
            "opacity": 1,
            "width": "60%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
        >
          Register Now
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
